modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -0,0 +1,403 @@
+ * ding_adgangsplatformen is not enabled.
+ */
+function ding_adgangsplatformen_preprocess_menu_link(&$vars) {
+  if (isset($vars['element']['#original_link']['menu_name']) && $vars['element']['#original_link']['menu_name'] == 'menu-tabs-menu') {
+    if ($vars['element']['#href'] == 'user') {
+      $vars['element']['#href'] = DING_ADGANGSPLATFORMEN_LOGIN_URL;
+    }
+  }
+}
+
+/**
+ * Implements hook_libraries_info().
+ *
+ * For defining external libraries.
+ */
+function ding_adgangsplatformen_libraries_info() {
+  return array(
+    'random_compat' => array(
+      'name' => 'random_compat',
+      'vendor url' => 'https://github.com/paragonie/random_compat',
+      'download url' => 'https://github.com/paragonie/random_compat',
+      'version' => 'v2.0.17',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+    'oauth2-client' => array(
+      'name' => 'oauth2-client',
+      'vendor url' => 'https://github.com/thephpleague/oauth2-client',
+      'download url' => 'https://github.com/thephpleague/oauth2-client',
+      'version' => '2.3.0',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+    'http-message' => array(
+      'name' => 'Common interface for HTTP messages',
+      'vendor url' => 'https://github.com/php-fig/http-message',
+      'download url' => 'https://github.com/php-fig/http-message',
+      'version' => '1.0.1',
+      'xautoload' => function ($adapter) {
+        $adapter->composerJson('composer.json');
+      },
+    ),
+  );
+}
+
+/**
+ * Implements hook_ding_provider_user().
+ *
+ * This module requires that the provider has authentication support.
+ */
+function ding_adgangsplatformen_ding_provider_user() {
+  return array(
+    'auth' => array(
+      'required' => TRUE,
+      'install time setup' => FALSE,
+    ),
+  );
+}
+
+/**
+ * Implements hook_ajax_login_command().
+ */
+function ding_adgangsplatformen_ajax_login_command($path) {
+  // Javascript need to do the redirect is loaded in the init hook.
+  ctools_include('ajax');
+
+  // Get authentication url and set the path to be redirected back to.
+  $url = ding_adgangsplatformen_login(FALSE, $path);
+
+  // Store the ajax callback url that triggered the ajax auth redirect.
+  $parameters = $_GET;
+  unset($parameters['q']);
+  $_SESSION['oauth2ajax'] = $_GET['q'] . '?' . drupal_http_build_query($parameters);
+
+  return ctools_ajax_command_redirect($url);
+}
+
+/**
+ * Get the authentication configuration.
+ *
+ * @return array
+ *   With the default parameters if not overridden in the administration
+ *   settings form.
+ */
+function ding_adgangsplatformen_get_configuration() {
+  global $base_url;
+  return variable_get('ding_adgangsplatformen_settings', array(
+    'clientId' => '',
+    'clientSecret' => '',
+    'urlAuthorize' => 'https://oauth.login.bib.dk/oauth/authorize',
```

Comment:
I think login.bib.dk should be used as URL and not oauth.login.bib.dk - better consult with the DBC team about that!

Reply:
They have changed it since this was developed :-)